tools/osbuilder/Makefile

```diff
@@ -23,9 +23,9 @@ TARGET_IMAGE          := $(IMAGES_BUILD_DEST)/kata-containers.img
 TARGET_INITRD         := $(IMAGES_BUILD_DEST)/kata-containers-initrd.img
 
 VERSION_FILE   := ./VERSION
-VERSION        := $(shell grep -v ^\# $(VERSION_FILE))
+VERSION        := $(if $(shell test -f $(VERSION_FILE)),$(shell grep -v ^\# $(VERSION_FILE)),"unknown")
 COMMIT_NO      := $(shell git rev-parse HEAD 2> /dev/null || true)
-COMMIT         := $(if $(shell git status --porcelain --untracked-files=no),${COMMIT_NO}-dirty,${COMMIT_NO})
+COMMIT         := $(if $(shell git status --porcelain --untracked-files=no 2> /dev/null),${COMMIT_NO}-dirty,${COMMIT_NO})
```

Comment:
For those 2, could we actually check whether it's a git repo as the first thing?  Even if we end up calling a helper shell script for doing that.

Reply:
I believe there is even two aspects here when we think of more production-like environments that would build the UVM components:
- the source code may not be a git repository
- git may not even be available

So, I guess one may want to check for both and that would be in `utils.mk`?

Reply:
So, let's break this down.

* Source code **is** a git repository
  * git **is** available, as a `git clone ...` happened
* Source code **is NOT** a git repository
  * git is available, but cannot be used to get the commit as this is not a git repo
  * git is not available

So, IMHO, it having git installed or not is really relevant if the source code is not a git repo.  And if it's a git repo, we know git is there as it was cloned from somewhere.

With that said, I'd just check whether the source code is a git repo or not.   And yes, I think the check could be done on `utils.mk`.

Reply:
I have realized the same pattern exists in various other Makefiles and aligned the calls consistently. In a subsequent change one could de-duplicate those calls across makefiles.